models/core/swaps.yml

```diff
@@ -0,0 +1,82 @@
+
+version: 2
+
+models:
+  - name: swaps
+    description: "Harmony Swaps"
+
+    columns:
+      - name: block_timestamp
+        description: When the trade was made.
+        tests:
+          - not_null
+
+      - name: log_id
+        description: The unique id.
+        tests:
+          - not_null
+          - unique
+
+      - name: tx_hash
+        description: The transaction hash.
+        tests:
+          - not_null
+
+      - name: pool_address
+        description: The pool address that the swap was conducted in.
+        tests:
+          - not_null
+
+      - name: token0_address
+        description: The `token0` address.
+        tests:
+
+      - name: token0_name
+        description: The `token0` name.
+        tests:
+
+      - name: token0_symbol
+        description: The `token0` symbol.
+        tests:
+
+      - name: amount0In
```

Comment:
kind of a noob question here, but in this form of swap, is the amount0Out (from_address) gonna be equal to amount0In (to_address)? 
and the same applies to token 1?

Reply:
Swaps are omni-directional.. so..
- amount0In + amount1Out = user gave token0 for token1
- amount1In + amount0Out = user gave token1 for token0
I haven't seen a mixed cased where all 4 numbers are used yet..  (but technically possible) 
The 0 and 1 are referring to the tokens, so e.g WETH-USDC pool, it'll never be the same (because of decimals, and WETH is worth way more then USDC)